src/psyclone/transformations.py

```diff
@@ -2825,15 +2825,28 @@ def apply(self, sched, options=None):
         elif isinstance(sched, GOInvokeSchedule):
             from psyclone.gocean1p0 import GOACCEnterDataDirective as \
                 AccEnterDataDir
+        elif isinstance(sched, NemoInvokeSchedule):
+            from psyclone.nemo import NemoACCEnterDataDirective as \
+                AccEnterDataDir
         else:
             # Should not get here provided that validate() has done its job
             raise InternalError(
                 f"ACCEnterDataTrans.validate() has not rejected an "
                 f"(unsupported) schedule of type {type(sched)}")
 
+        # Determine where to place the directive
```

Comment:
Please extend this comment to explain a bit about the algorithm used - i.e. what is driving the choice of location for the directive?

Reply:
Done

Reply:
It does have the advantage of potentially being placed after an update directive, potentially avoiding the copies incurred by the latter.